source/event_loop.c

```diff
@@ -532,3 +569,131 @@ struct aws_event_loop *aws_event_loop_new_base(
 
     return event_loop;
 }
+
+/**
+ * Override default event loop type. Only used internally in tests.
+ *
+ * If the defined type is not supported on the current platform, the event loop type would reset to
+ * AWS_EVENT_LOOP_PLATFORM_DEFAULT.
+ */
+void aws_event_loop_override_default_type(enum aws_event_loop_type default_type_override) {
+    if (aws_event_loop_type_validate_platform(default_type_override) == AWS_OP_SUCCESS) {
+        s_default_event_loop_type_override = default_type_override;
+    } else {
+        s_default_event_loop_type_override = AWS_EVENT_LOOP_PLATFORM_DEFAULT;
+    }
+}
+
+/**
+ * Return the default event loop type. If the return value is `AWS_EVENT_LOOP_PLATFORM_DEFAULT`, the function failed to
+ * retrieve the default type value.
+ * If `aws_event_loop_override_default_type` has been called, return the override default type.
+ */
+static enum aws_event_loop_type aws_event_loop_get_default_type(void) {
+    if (s_default_event_loop_type_override != AWS_EVENT_LOOP_PLATFORM_DEFAULT) {
+        return s_default_event_loop_type_override;
+    }
+/**
+ * Ideally we should use the platform definition (e.x.: AWS_OS_APPLE) here, however the platform
+ * definition was declared in aws-c-common. We probably do not want to introduce extra dependency here.
+ */
+#ifdef AWS_ENABLE_KQUEUE
+    return AWS_EVENT_LOOP_KQUEUE;
+#endif
+#ifdef AWS_ENABLE_DISPATCH_QUEUE
+    return AWS_EVENT_LOOP_DISPATCH_QUEUE;
+#endif
+#ifdef AWS_ENABLE_EPOLL
+    return AWS_EVENT_LOOP_EPOLL;
+#endif
+#ifdef AWS_OS_WINDOWS
+    return AWS_EVENT_LOOP_IOCP;
+#endif
+}
+
+static int aws_event_loop_type_validate_platform(enum aws_event_loop_type type) {
+    switch (type) {
+        case AWS_EVENT_LOOP_EPOLL:
+#ifndef AWS_ENABLE_EPOLL
+            AWS_LOGF_DEBUG(AWS_LS_IO_EVENT_LOOP, "Event loop type EPOLL is not supported on the platform.");
```

Comment:
I’d like to clarify the distinction between debug and error logs to ensure we're aligned:

For this case, since we're raising and returning an error for unsupported platforms, the caller can handle the returned value and process the result accordingly. I feel this would fall under a debug log.

On the other hand, I think an error log applies to situations that cannot be recovered from. For example, in the `aws_event_loop_get_default_type(void`)` function you mentioned above, if nothing can be returned, that would definitely qualify as an error log.